app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -1022,15 +1022,17 @@ public void updateTabLayoutVisibility() {
             // call `post()` to be sure `viewPager.getHitRect()`
             // is up to date and not being currently recomputed
             binding.tabLayout.post(() -> {
-                if (getContext() != null) {
+                final Context context = getContext();
+                if (context != null) {
                     final Rect pagerHitRect = new Rect();
                     binding.viewPager.getHitRect(pagerHitRect);
 
-                    final Point displaySize = new Point();
-                    Objects.requireNonNull(ContextCompat.getSystemService(getContext(),
-                            WindowManager.class)).getDefaultDisplay().getSize(displaySize);
+                    final WindowManager windowManager = Objects.requireNonNull(ContextCompat
+                            .getSystemService(context, WindowManager.class));
+                    final DisplayCompat.ModeCompat mode = DisplayCompat.getMode(context,
+                            windowManager.getDefaultDisplay());
 
-                    final int viewPagerVisibleHeight = displaySize.y - pagerHitRect.top;
+                    final int viewPagerVisibleHeight = mode.getPhysicalHeight() - pagerHitRect.top;
```

Comment:
This is not correct. `viewPagerVisibleHeight` should represent the height of the view pager that is visible on the screen, but here the size of Android's bottom bar is also summed, since `getPhysicalHeight()` is used. But now that I see it, it seems like it was also wrong before, since `getDefaultDisplay().getSize()` returned the height of the app and the above difference was also wrong. What `viewPagerVisibleHeight` should be is "y_of_the_bottom_of_the_activity (i.e. just above Android's bottom bar) - y_of_the_top_of_viewPager". Is there a way to do this with the compat library? I saw that `getDefaultDisplay().getSize()` was deprecated in API 30 in favour of `DisplayMetrics`, and `WindowMetrics#getBounds()` would provide us with "y_of_the_bottom_of_the_activity", but that method has not yet been backported in the compat library.

Reply:
There's a [Window Manager](https://developer.android.com/jetpack/androidx/releases/window) library, but it's still in alpha.